src/levels.jl

```diff
@@ -22,11 +50,73 @@ Base.:(==)(l1::Level, l2::Level) = ((l1.csf == l2.csf) && (l1.J == l2.J))
 Base.isless(l1::Level, l2::Level) = ((l1.csf < l2.csf)) ||
     ((l1.csf == l2.csf)) && (l1.J < l2.J)
 
+"""
+    J_range(term::Term)
+
+List the permissible values of the total angular momentum ``J`` for
+`term`.
+
+# Examples
+
+```jldoctest
+julia> J_range(T"¹S")
+0:0
+
+julia> J_range(T"²S")
+1/2:1/2
+
+julia> J_range(T"³P")
+0:2
+
+julia> J_range(T"²D")
+3/2:5/2
+```
+"""
 J_range(term::Term) = abs(term.L-term.S):(term.L+term.S)
-J_range(term::HalfInteger) = term:term
 
+"""
+    J_range(J)
+
+The permissible range of the total angular momentum ``J`` in ``jj``
+coupling is simply the value of ``J`` for the final term.
+"""
+J_range(J::HalfInteger) = J:J
+J_range(J::Integer) = J_range(convert(HalfInteger, J))
+
+"""
+    levels(csf)
+
+Generate all permissible [`Level`](@ref)s given `csf`.
+
+# Examples
+
+```jldoctest
+julia> csfs(c"1s 2p")
+2-element Vector{NonRelativisticCSF{Orbital{Int64}, Seniority}}:
+ 1s(₁²S|²S) 2p(₁²Pᵒ|¹Pᵒ)-
+ 1s(₁²S|²S) 2p(₁²Pᵒ|³Pᵒ)-
+
+julia> csfs(rc"1s 2p")
+2-element Vector{RelativisticCSF{RelativisticOrbital{Int64}, Seniority}}:
+ 1s(₁1/2|1/2) 2p(₁3/2|1)-
+ 1s(₁1/2|1/2) 2p(₁3/2|2)-
+
+julia> csfs(rc"1s 2p-")
+2-element Vector{RelativisticCSF{RelativisticOrbital{Int64}, Seniority}}:
+ 1s(₁1/2|1/2) 2p-(₁1/2|0)-
+ 1s(₁1/2|1/2) 2p-(₁1/2|1)-
+```
```

Comment:
These doctests don't actually call `levels`?

Reply:
Should be fixed now